source/state_representation/include/state_representation/parameters/Parameter.hpp

```diff
@@ -120,11 +125,25 @@ inline void Parameter<T>::set_value(const T& value) {
   this->value_ = value;
 }
 
+/**
+ * @brief Create a Parameter object that is owned by a shared_ptr
+ * @tparam T The type of the value stored in the parameter
+ * @param name The name of the parameter
+ * @param param_value The value of the parameter
+ * @return A shared_ptr that owns the newly created Parameter object
+ */
 template<typename T>
 static std::shared_ptr<Parameter<T>> make_shared_parameter(const std::string& name, const T& param_value) {
   return std::make_shared<Parameter<T>>(name, param_value);
 }
 
+/**
+ * @brief TODO
+ * @param name The name of the parameter
+ * @param type The type of the parameter
+ * @param parameter_state_type The state type of the parameter, if applicable
+ * @return TODO
```

Comment:
I would suggest something along these lines:
```suggestion
 * @brief Encapsulate a parameter in a ParameterInterface pointer
 * @param name The name of the parameter
 * @param type The type of the parameter
 * @param parameter_state_type The state type of the parameter, if applicable
 * @return A ParameterInterface pointer holding a reference to a Parameter object
```